admin-dev/themes/new-theme/js/pages/product/components/dropzone/Dropzone.vue

```diff
@@ -330,9 +342,23 @@
         });
 
         this.dropzone.on(DropzoneEvents.addedFile, (file: PSDropzoneFile) => {
+          // Be aware the file type is not always a PSDropzoneFile.
+          // this event is called multiple times and the file type may differ:
+          // 1. when file is being uploaded it is some sort of native File type
+          // 2. when its being loaded from api it is the PSDropzoneFile type
+
           file.previewElement.dataset.id = file.image_id;
 
-          if (file.is_cover) {
+          // check if image is associated with current shop.
+          // If shop_ids is missing, it means the image is being uploaded, therefore it is not PSDropzoneFile type,
+          // so it will be associated to the shop by default
+          file.isAssociatedToCurrentShop = !file.shop_ids || file.shop_ids.includes(this.shopId);
```

Comment:
When it's empty you should maybe populate the shop_ids field:
```suggestion
          if (!file.shop_ids) {
            file.shop_ids = [this.shopId];
          }
          file.isAssociatedToCurrentShop = file.shop_ids.includes(this.shopId);
```